localstack/services/ec2/resource_providers/aws_ec2_securitygroup.py

```diff
@@ -0,0 +1,149 @@
+# LocalStack Resource Provider Scaffolding v2
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Optional, TypedDict
+
+import localstack.services.cloudformation.provider_utils as util
+from localstack.services.cloudformation.resource_provider import (
+    OperationStatus,
+    ProgressEvent,
+    ResourceProvider,
+    ResourceRequest,
+)
+
+
+class EC2SecurityGroupProperties(TypedDict):
+    GroupDescription: Optional[str]
+    GroupId: Optional[str]
+    GroupName: Optional[str]
+    Id: Optional[str]
+    SecurityGroupEgress: Optional[list[Egress]]
+    SecurityGroupIngress: Optional[list[Ingress]]
+    Tags: Optional[list[Tag]]
+    VpcId: Optional[str]
+
+
+class Ingress(TypedDict):
+    IpProtocol: Optional[str]
+    CidrIp: Optional[str]
+    CidrIpv6: Optional[str]
+    Description: Optional[str]
+    FromPort: Optional[int]
+    SourcePrefixListId: Optional[str]
+    SourceSecurityGroupId: Optional[str]
+    SourceSecurityGroupName: Optional[str]
+    SourceSecurityGroupOwnerId: Optional[str]
+    ToPort: Optional[int]
+
+
+class Egress(TypedDict):
+    IpProtocol: Optional[str]
+    CidrIp: Optional[str]
+    CidrIpv6: Optional[str]
+    Description: Optional[str]
+    DestinationPrefixListId: Optional[str]
+    DestinationSecurityGroupId: Optional[str]
+    FromPort: Optional[int]
+    ToPort: Optional[int]
+
+
+class Tag(TypedDict):
+    Key: Optional[str]
+    Value: Optional[str]
+
+
+REPEATED_INVOCATION = "repeated_invocation"
+
+
+class EC2SecurityGroupProvider(ResourceProvider[EC2SecurityGroupProperties]):
+
+    TYPE = "AWS::EC2::SecurityGroup"  # Autogenerated. Don't change
+    SCHEMA = util.get_schema_path(Path(__file__))  # Autogenerated. Don't change
+
+    def create(
+        self,
+        request: ResourceRequest[EC2SecurityGroupProperties],
+    ) -> ProgressEvent[EC2SecurityGroupProperties]:
+        """
+        Create a new resource.
+
+        Primary identifier fields:
+          - /properties/Id
+
+        Required properties:
+          - GroupDescription
+
+        Create-only properties:
+          - /properties/GroupDescription
+          - /properties/GroupName
+          - /properties/VpcId
+
+        Read-only properties:
+          - /properties/Id
+          - /properties/GroupId
+
+
+
+        """
+        model = request.desired_state
+        ec2 = request.aws_client_factory.ec2
+
+        params = util.select_attributes(model, ["GroupName", "VpcId", "GroupDescription"])
+
+        if not params.get("GroupName"):
+            params["GroupName"] = util.generate_default_name(
+                request.stack_name, request.logical_resource_id
+            )
+        response = ec2.create_security_group(
+            VpcId=params["VpcID"],
+            GroupName=params["GroupName"],
+            Description=params.get("GroupDescription", ""),
+        )
+        model["GroupId"] = response["GroupId"]
```

Comment:
The primary identifier is `Id` so this needs to be added as well

Reply:
Fixed